apps/frontend/core-dapp/src/features/history/HistoryTransaction.tsx

```diff
@@ -0,0 +1,103 @@
+import React from 'react'
+import { Flex, Icon, spacingIncrement } from 'prepo-ui'
+import styled from 'styled-components'
+import { observer } from 'mobx-react-lite'
+import Link from 'next/link'
+import { NETWORKS } from 'prepo-constants'
+import { LinkProps } from 'next/dist/client/link'
+import { HistoryTransaction as HistoryTransactionEntity } from './history.types'
+import { buttonColors as transactionTypeColors } from './History'
+import { numberFormatter } from '../../utils/numberFormatter'
+import PositionLabel from '../position/PositionLabel'
+import { getFullDateShortenMonthFromSeconds } from '../../utils/date-utils'
+import { useRootStore } from '../../context/RootStoreProvider'
+import { PositionName } from '../position/Position'
+
+const { toUsd } = numberFormatter
+
+const Label = styled.p`
+  color: ${({ theme }): string => theme.color.neutral1};
+  margin: 0;
+`
+
+const EventLink = styled.a<{ $event: string }>`
+  align-items: center;
+  background: ${({ $event, theme }): string =>
+    theme.color[transactionTypeColors[$event].backgroundColor]};
+  border-radius: ${spacingIncrement(8)};
+  color: ${({ $event, theme }): string => theme.color[transactionTypeColors[$event].color]};
+  display: flex;
+  padding: ${spacingIncrement(2)} ${spacingIncrement(8)};
+`
+
+type HistoryTransactionProps = {
+  event: HistoryTransactionEntity['event']
+  eventType: HistoryTransactionEntity['eventType']
+  iconName: HistoryTransactionEntity['iconName']
+  marketId: HistoryTransactionEntity['marketId']
+  name: HistoryTransactionEntity['name']
+  timestamp: HistoryTransactionEntity['timestamp']
+  transactionHash: HistoryTransactionEntity['transactionHash']
+  usdValue: HistoryTransactionEntity['usdValue']
+}
+const HistoryTransaction: React.FC<HistoryTransactionProps> = ({
+  event,
+  eventType,
+  iconName,
+  marketId,
+  name,
+  timestamp,
+  transactionHash,
+  usdValue,
+}) => {
+  const { web3Store } = useRootStore()
+  const { network } = web3Store
+
+  let nameRedirectUrl: LinkProps['href'] | undefined
+  if (marketId !== undefined && (event === 'Opened' || event === 'Closed'))
+    nameRedirectUrl = {
+      pathname: '/trade',
+      query: {
+        action: event === 'Opened' ? 'open' : 'close',
+        direction: eventType,
+        marketId,
+      },
+    }
+  if (event === 'Deposited') nameRedirectUrl = '/portfolio/deposit'
+  if (event === 'Withdrawn') nameRedirectUrl = '/portfolio/withdraw'
```

Comment:
I guess there isn't a case where nameRedirectUrl would be undefined + since event won't change for each transaction, we can do this in a memo so the logic doesn't get computed again:

```suggestion
  const nameRedirectUrl = useMemo(() => {
    switch (event) {
      case 'Deposited':
        return '/portfolio/deposit'
      case 'Withdrawn':
        return '/portfolio/withdraw'
      default:
        return {
          pathname: '/trade',
          query: { action: event === 'Opened' ? 'open' : 'close', direction: eventType, marketId },
        }
    }
  }, [event, eventType, marketId])
```

Reply:
I agree with `useMemo` because it looks cleaner (it's its own function, so we can use return to alter the control flow), but not for the performance optimization. I think that would be premature optimization.

This is one of the examples why Kotlin is my favorite language, control statements are expressions so you can assign them to a variable 😆 

```kt
val nameRedirectUrl = when (event) {
  "Deposited" -> "/portfolio/deposit"
  "Withdrawn" -> "/portfolio/withdrawn"
  else -> // the location object
}
```